src/fileimpl.cpp

```diff
@@ -121,6 +121,16 @@ namespace zim
         log_fatal("last offset (" << lastOffset << ") larger than file size (" << zimFile->fsize() << ')');
         throw ZimFileFormatError("last cluster offset larger than file size; file corrupt");
       }
+
+      auto nb_articles = getCountArticles().v;
+      articleListByCluster.reserve(nb_articles);
+
+      for(zim::article_index_type i = 0; i < nb_articles; i++)
+      {
+          articleListByCluster.push_back(std::make_pair(getDirent(article_index_t(i))->getClusterNumber().v, i));
+      }
+
+      std::sort(articleListByCluster.begin(), articleListByCluster.end());
```

Comment:
It would better to avoid to create this list every time. It is pretty time consuming and most of the time, we don't need it. We should generate it only if (when) we need it (when we use `getArticleByClusterOrder`.
(But we need to be thread safe, so we must protect the list creation with mutex).